graylog2-server/src/main/java/org/graylog2/configuration/RunsWithDataNodeDiscoveryProvider.java

```diff
@@ -57,24 +44,6 @@ public Boolean get() {
     }
 
     private Boolean doGet() {
-        final PreflightConfigResult preflightResult = preflightConfigService.getPreflightConfigResult();
-
-        // if preflight is finished, we assume that there will be some datanode registered via node-service.
-        if (preflightResult == PreflightConfigResult.FINISHED) {
-            final List<URI> discovered = discover();
-            if (!discovered.isEmpty()) {
-                LOG.debug("Running with DataNode(s).");
-                return Boolean.TRUE;
-            }
-        }
-        LOG.debug("Not running with DataNode(s).");
-        return Boolean.FALSE;
-    }
-
-    private List<URI> discover() {
-        return nodeService.allActive().values().stream()
-                .map(Node::getTransportAddress)
-                .map(URI::create)
-                .collect(Collectors.toList());
+        return !nodeService.allActive().isEmpty();
```

Comment:
would it make sense to add an additional filter for `DataNodeStatus.AVAILABLE` here?